sdks/go/examples/stringsplit/stringsplit.go

```diff
@@ -0,0 +1,249 @@
+// See ProcessElement for more details.
+type StringSplitFn struct {
+	BufSize int64
+}
+
+// CreateInitialRestriction creates an offset range restriction for each element
+// with the size of the restriction corresponding to the length of the string.
+func (fn *StringSplitFn) CreateInitialRestriction(s string) offsetrange.Restriction {
+	rest := offsetrange.Restriction{Start: 0, End: int64(len(s))}
+	log.Debugf(context.Background(), "StringSplit CreateInitialRestriction: %v", rest)
+	return rest
+}
+
+// SplitRestriction performs initial splits so that each restriction is split
+// into 5.
+func (fn *StringSplitFn) SplitRestriction(s string, rest offsetrange.Restriction) []offsetrange.Restriction {
+	size := rest.End - rest.Start
+	splitPts := []int64{
+		rest.Start,
+		rest.Start + (size / 5),
+		rest.Start + (size * 2 / 5),
+		rest.Start + (size * 3 / 5),
+		rest.Start + (size * 4 / 5),
+		rest.End,
+	}
+	var splits []offsetrange.Restriction
+	for i := 0; i < len(splitPts)-1; i++ {
+		splits = append(splits, offsetrange.Restriction{Start: splitPts[i], End: splitPts[i+1]})
+	}
+	log.Debugf(context.Background(), "StringSplit SplitRestrictions: %v -> %v", rest, splits)
+	return splits
+}
+
+// RestrictionSize returns the size as the difference between the restriction's
+// start and end.
+func (fn *StringSplitFn) RestrictionSize(s string, rest offsetrange.Restriction) float64 {
+	size := float64(rest.End - rest.Start)
+	log.Debugf(context.Background(), "StringSplit RestrictionSize: %v -> %v", rest, size)
+	return size
+}
+
+// CreateTracker creates an offset range restriction tracker out of the offset
+// range restriction.
+func (fn *StringSplitFn) CreateTracker(rest offsetrange.Restriction) *offsetrange.Tracker {
+	return offsetrange.NewTracker(rest)
+}
+
+// ProcessElement splits a string into substrings of a specified size (set in
+// StringSplitFn.BufSize).
+//
+// Note that the substring blocks are not guaranteed to line up with the
+// restriction boundaries. ProcessElement is expected to emit any substring
+// block that begins in its restriction, even if it extends past the end of the
+// restriction.
+//
+// Example: If BufSize is 100, then a restriction of 75 to 325 should emit the
+// following substrings: [100, 200], [200, 300], [300, 400]
+func (fn *StringSplitFn) ProcessElement(rt *offsetrange.Tracker, elem string, emit func(string)) error {
+	log.Debugf(context.Background(), "StringSplit ProcessElement: Tracker = %v", rt)
+	i := rt.Rest.Start
+	if rem := i % fn.BufSize; rem != 0 {
+		i += fn.BufSize - rem // Skip to next multiple of BufSize.
+	}
+	strEnd := int64(len(elem))
+
+	for ok := rt.TryClaim(i); ok == true; ok = rt.TryClaim(i) {
+		if i+fn.BufSize > strEnd {
+			emit(elem[i:])
+		} else {
+			emit(elem[i : i+fn.BufSize])
+		}
+		i += fn.BufSize
+	}
+	// TODO(BEAM-9799): Remove this check once error checking is automatic.
+	if err := rt.GetError(); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Consider just returning rt.GetError() instead of the extra block. 
Having the rt.GetError() automatically checked is feasible given we already have the tracker pointer. We can just 

```
if ge, ok := rt.(interface{GetError() error}); ok {
  err := ge.GetError()
 ... handle error...
}
```

Reply:
Done. I'll hold off on having it automatically checked for the moment, just because I wanna add a test along with that when I get to it.